Source/Particles/Collision/BackgroundMCCCollision.cpp

```diff
@@ -0,0 +1,382 @@
+                              amrex::ParticleReal ua_x, ua_y, ua_z;
+                              amrex::ParticleReal uCOM_x, uCOM_y, uCOM_z;
+
+                              // get velocities of gas particles from a Maxwellian distribution
+                              ua_x = vel_std * amrex::RandomNormal(0_rt, 1.0_rt, engine);
+                              ua_y = vel_std * amrex::RandomNormal(0_rt, 1.0_rt, engine);
+                              ua_z = vel_std * amrex::RandomNormal(0_rt, 1.0_rt, engine);
+
+                              // calculate the center of momentum velocity
+                              uCOM_x = (mass1 * ux[ip] + mass_a * ua_x) / (mass1 + mass_a);
+                              uCOM_y = (mass1 * uy[ip] + mass_a * ua_y) / (mass1 + mass_a);
+                              uCOM_z = (mass1 * uz[ip] + mass_a * ua_z) / (mass1 + mass_a);
+
+                              // calculate relative velocity of collision and collision energy if
+                              // the colliding particle is an ion. For electron collisions we
+                              // cannot use the relative velocity since that allows the
+                              // possibility where the electron kinetic energy in the lab frame
+                              // is insufficient to cause excitation but not in the COM frame -
+                              // for energy to balance this situation requires the neutral to
+                              // lose energy during the collision which we don't currently
+                              // account for.
+                              if (mass_a / mass1 > 1e3) {
+                                  v_coll2 = ux[ip]*ux[ip] + uy[ip]*uy[ip] + uz[ip]*uz[ip];
+                                  E_coll = 0.5_rt * mass1 * v_coll2 / PhysConst::q_e;
+                              }
+                              else {
+                                  v_coll2 = (
+                                             (ux[ip] - ua_x)*(ux[ip] - ua_x)
+                                             + (uy[ip] - ua_y)*(uy[ip] - ua_y)
+                                             + (uz[ip] - ua_z)*(uz[ip] - ua_z)
+                                             );
+                                  E_coll = (
+                                            0.5_rt * mass1 * mass_a / (mass1 + mass_a) * v_coll2
+                                            / PhysConst::q_e
+                                            );
+                              }
+                              v_coll = sqrt(v_coll2);
+
+                              // loop through all collision pathways
+                              for (size_t i = 0; i < process_count; i++) {
+                                  auto const& scattering_process = **(scattering_processes + i);
+
+                                  // get collision cross-section
+                                  sigma_E = scattering_process.getCrossSection(E_coll);
```

Comment:
@RemiLehe This call site is where the conversion to a free function and avoiding MCCProcess instances living in device memory runs into trouble

Reply:
We discussed this on Slack, and agreed that it would be a bigger hassle than immediately worthwhile to eliminate managed memory usage here.